pkg/ctl/install/cmd.go

```diff
@@ -0,0 +1,100 @@
+package install
+
+import (
+	"context"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+// Command will create the `install` commands
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	verbCmd := cmdutils.NewVerbCmd("install", "Install components in a cluster", "")
+
+	cmdutils.AddResourceCmd(flagGrouping, verbCmd, installFluxCmd)
+
+	return verbCmd
+}
+
+type installFluxOpts struct {
+	gitURL              string
+	gitBranch           string
+	gitPaths            []string
+	gitLabel            string
+	gitUser             string
+	gitEmail            string
+	gitFluxPath         string
+	namespace           string
+	tillerNamespace     string
+	timeout             time.Duration
+	amend               bool
+	noHelmOp            bool
+	noTiller            bool
+	tillerHost          string
+	helmOpTLSCertFile   string
+	helmOpTLSKeyFile    string
+	helmOpTLSCACertFile string
+}
+
+func installFluxCmd(cmd *cmdutils.Cmd) {
+	cmd.ClusterConfig = api.NewClusterConfig()
+	cmd.SetDescription(
+		"flux",
+		"Bootstrap Flux, installing it in the cluster and initializing its manifests in the specified Git repository",
+		"",
+	)
+	var opts installFluxOpts
+	cmd.SetRunFuncWithNameArg(func() error {
+		installer, err := newFluxInstaller(context.Background(), cmd, &opts)
+		if err != nil {
+			return err
+		}
+		return installer.run(context.Background())
+	})
+	cmd.FlagSetGroup.InFlagSet("Flux installation", func(fs *pflag.FlagSet) {
+		fs.StringVar(&opts.gitURL, "git-url", "",
+			"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
+		fs.StringVar(&opts.gitBranch, "git-branch", "master",
+			"Git branch to be used by Flux")
+		fs.StringSliceVar(&opts.gitPaths, "git-paths", []string{},
+			"Relative paths within the Git repo for Flux to locate Kubernetes manifests")
+		fs.StringVar(&opts.gitLabel, "git-label", "flux",
+			"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+		fs.StringVar(&opts.gitUser, "git-user", "Flux",
+			"Username to use as Git committer")
+		fs.StringVar(&opts.gitEmail, "git-email", "",
+			"Email to use as Git committer")
+		fs.StringVar(&opts.gitFluxPath, "git-flux-subdir", "flux/",
+			"Directory within the Git repository where to commit the Flux manifests")
+		fs.StringVar(&opts.namespace, "namespace", "flux",
+			"Cluster namespace where to install Flux and the Helm Operator")
+		fs.StringVar(&opts.tillerNamespace, "tiller-namespace", "kube-system",
+			"Cluster namespace where to install Tiller")
+		fs.BoolVar(&opts.noHelmOp, "no-helmop", false,
+			"Do not install the Flux Helm Operator (implies --no-tiller)")
+		fs.BoolVar(&opts.noTiller, "no-tiller", false,
```

Comment:
Let's avoid separate flag, it's either helm and the operator or neither.

Reply:
I invoke @kalbir for this one as well

Reply:
I don't understand the pro/con of the options, can someone outline what happens in each case?

Reply:
It's related to https://github.com/weaveworks/eksctl/pull/1136#discussion_r313822189 . This flag makes the installation of Tiller optional (in case Tiller was already installed in the cluster).

Reply:
I guess if we put everything into a `flux` namespace we can assume that the user doesn't have a tiller in that namespace (since we are creating it). So that should get us around the case where a user already has tiller installed in say `kube-system`?

So it seems to me sensible to put tiller in `flux` and then we have less difficulty dealing with pre-existing tillers. Also as we are encouraging the user to do everything through gitops so the I think we have less worry about them having to supply a `tiller-namespace=flux` (though we should put that in the documentation somewhere.

Does that make sense?

Reply:
> So that should get us around the case where a user already has tiller installed in say `kube-system`?

As a user I may not be satisfied to have two Tillers since I may already have machinery in place to deal with one Tiller, where I installed it. However, we could probably consider that a corner case?

Reply:
I think we are talking about the thing multiple threads - how opinionated the flux-tiller installation should be. My point being that it needs to only expose essential options to begin with, any customisation can be added later (when users request it).
Also, we have a strategy to keep less flags, and anything extra only gets exposed via config file.
I do think we need to join threads and agree on something, and that doesn't have to happen before this PR gets merged. I am mostly commenting on what I am seeing, this is an experimental feature and will have to go through an overall review before we graduate it.